### PR TITLE
Suggestion

### LINKED ISSUE #4020: Baritone only moves pickaxes and throwaway blocks to the hot bar and does that even if no process is active
Inventory behavior only picks tools of the pickaxe type into the hotbar. If your shovel breaks and you dont have a second one in your hotbar baritone will use its fist to mine dirt blocks instead of using the backup shovel in your inventory regardless of your allowInventory setting.
Also it will always move pickaxes and throwAwayItems to the hotbar. maybe there is a good reason for that but i think if baritone doesnt have a process running it shouldnt move itmes in the inventory.
https://github.com/cabaletta/baritone/blob/1ea071de09631a41f0ea743df036db8000b7147c/src/main/java/baritone/behavior/InventoryBehavior.java#L60-L66

### PR DESCRIPTION
**Pull request by rycbar0**
a suggestion on how to fix #4020

### CODE REVIEW

File: src/main/java/baritone/behavior/InventoryBehavior.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public InventoryBehavior(Baritone baritone) {
 
     @Override
     public void onTick(TickEvent event) {
-        if (!Baritone.settings().allowInventory.value) {
+        if (!Baritone.settings().allowInventory.value || baritone.getPathingControlManager().mostRecentInControl().orElse(null) == null) {
```
Comment (ZeroMemes) [inline src/main/java/baritone/behavior/InventoryBehavior.java:49]: This can be replaced with `isPresent()`

**Comment by ZacSharp** [comment]
This does certainly fix both issues (you can also add a "fix" or "fixes" for #3000 and #3409), but I don't like the way you fixed the "only pickaxes are moved" problem.

To fix the "don't move items you don't need" issue properly we have to move items only when we know we actually need them, because otherwise this fix is just going to result in follow up bug reports where people complain it insists on having a pickaxe even though `allowBreak` is off or it moves tools even though no blocks in `allowBreakAnyway` are in render distance or it moves stone with `allowPlace` disabled or it doesn't choose the correct super-hyper-spade from whatever-mod because that spade only works for blocks from whatever-mod or it doesn't move shears to hotbar to break leaves (same for hoes and swords).
What we need is a general purpose item selection method like `InventoryBehavior.throwaway` which can also select items by priority. The intention behind using general priority based selection rather than specialized tool selection is that e.g. `ElytraBehavior` also have use for this. It currently prioritizes boosting fireworks by trying to get some and if that fails it tries to get fireworks without boost. Prioritizing longer boosts would currently require two more queries (because there's 3 boost lengths) or duplicating the logic.
.

**Comment by ZacSharp** [comment]
Just noticed I said that I didn't like one fix and then went on about the other, oops.
Guess that's because I really want on demand selection, which is a fix to both the hotbar space problem (3 tools + throwaway is 4 slots before it even started working on a task) and the various cases where it still won't move the tool or still would move the tool way too eagerly (i.e. if it can't possibly need it).

**Comment by rycbar0** [comment]
no longer relevant

**Pull request closed without merging**